src/mgr/MgrClient.h

```diff
@@ -104,7 +104,7 @@ class MgrClient : public Dispatcher
   bool handle_command_reply(MCommandReply *m);
 
   void send_pgstats();
-  void set_pgstats_cb(std::function<MPGStats*()> cb_)
+  void set_pgstats_cb(const std::function<MPGStats*()> &cb_)
```

Comment:
i'd suggest pass by rvalue reference here. and move to pgstats_cb like
```c++
void set_pgstats_cb(std::function<MPGStats*()>&& cb_) 
{
  Mutex::Locker l(lock);
  pgstats_cb = std::move(cb_);
}
```

also, i'd recommend put the `&` right after the type of the parameter, as the `&` is a part of the type.

Reply:
I get your point but I'm not sure if I should create a new function of just modify that one. I think you want me to modify it, but I think I should create a new one.

The `*` is also next to the parameter names in that file. Should I move those too next to the type?

Reply:
> but I think I should create a new one.

could you explain the rationale behind this?

> Should I move those too next to the type?

i think it'd be better to just change the lines you are updating.

Reply:
I think I should create a new one because, as far as I know, for any class T
```cpp
void foo(T& p);
```
is not the same as
```cpp
void foo(T&& p);
```
but I should point out I was just reading about rvalue references -- I didn't know about them.

I figured we should get a compile-time error anyway, so I'll make your suggested changes, push them and stay tuned to the CI log.